### PR TITLE
[Opset] Initialization of OV Opset15

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
 -  Initialization of OV Opset15 
  (Discussed internally and agreed to introduce new opset in 2024.2)
  
  To provide more agile approach, this PR has been updated with an idea (proposed by @mmikolajcz) to include only new operators within the new Opset, and fill it up at the end of the development period (not necessarily per release).
This should lead to infrequent introduction of further opsets.
 
\* Currently common opsets tests rely on Parameter/ShapeOf/Convert in every opset, so they were added.
But this approach for opset testing should be revised. 
 
### Tickets:
 - 138268, 131683

### CODE REVIEW

**Review by rkazants — DISMISSED** [review]
Why do we need new opset?
I think new opset is needed when we update existing operation introduced in opset14.
Otherwise, it is not needed.

**Comment by mitruska** [comment]
> Why do we need new opset? I think new opset is needed when we update existing operation introduced in opset14. Otherwise, it is not needed.

New version of `ScatterNDUpdate` is going to be introduced.

**Comment by rkazants** [comment]
> > Why do we need new opset? I think new opset is needed when we update existing operation introduced in opset14. Otherwise, it is not needed.
> 
> New version of `ScatterNDUpdate` is going to be introduced.

Anyway new opset is not needed because `ScatterNDUpdate` is not updated in opset14

**Comment by mitruska** [comment]
> > > Why do we need new opset? I think new opset is needed when we update existing operation introduced in opset14. Otherwise, it is not needed.
> > 
> > 
> > New version of `ScatterNDUpdate` is going to be introduced.
> 
> Anyway new opset is not needed because `ScatterNDUpdate` has been updated in opset3

@rkazants, this assumption is not fully correct:
 "I think new opset is needed when we update existing operation introduced in opset14."

The goal of every new opset is to not overwrite/change the version of the operator within **any already released opset** (if the change can be not backward compatible). Opset14 contains `v3::ScatterNDUpdate`, and update of the version within the opset14 will result with different version produced by the same opset. 

To provide more agile approach, this PR has been updated with an idea (proposed by @mmikolajcz) to include only new operators within the new Opset, and fill it up at the end of the development period (not necessarily per release).
This should lead to infrequent introduction of further opsets.

File: src/bindings/python/src/openvino/runtime/opset15/ops.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        {
+            "strides": strides,
+            "pads_begin": pads_begin,
+            "pads_end": pads_end,
+            "kernel": kernel_shape,
+            "exclude-pad": exclude_pad,
+            "rounding_type": rounding_type.upper(),
+            "auto_pad": auto_pad.upper(),
+        },
+    )
```
[the remark this answers is not in the document]
Comment (mitruska) [reply src/bindings/python/src/openvino/runtime/opset15/ops.py]: Sure thing, it was a copy of the opset14:) Thanks!

**Review by mmikolajcz — APPROVED** [review]

**Review by praasz — APPROVED** [review]

**Review by PiotrKrzem — APPROVED** [review]

**Review by p-wysocki — APPROVED** [review]

**Pull request merged**